### ISSUE TITLE
False positive ``subprocess-run-check`` when passing check = True by kwargs

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [PyCQA/pylint#5363]
### Bug description

```python
command = ["myprogram"]
run_kwargs = {"check": True}
subprocess.run(command, **run_kwargs) # raise a subprocess-run-check
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
W1510: Using subprocess.run without explicitly set `check` is not recommended. (subprocess-run-check)
```

### Expected behavior

No warning.

### Pylint version

```shell
pylint 2.11.2-dev0
astroid 2.9.0
Python 3.8.10 (default, Sep 28 2021, 16:10:42) 
[GCC 9.3.0]
```

**Comment by clavedeluna** [comment]
Am I correct in concluding that this is an astroid fix? [in here](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/stdlib.py#L479) 

For a test example `subprocess.run(command, check=True)`  `node.keywords`  is `node.keywords[0].arg` is check` as expected, while for call like 
```
run_kwargs = {"check": True}
subprocess.run(command, **run_kwargs)
```
`node.keywords[0].arg` is None so we never enter the if statement to emit the message.

**Comment by Pierre-Sassoulas** [comment]
Sounds about right. I added the control-flow label originally, I don't remember why as there is no control flow involved here. Probably that the content of the dict is hard to predict as soon as you add conditional in its creation, for example:
```python

import subprocess

command = ["myprogram"]
if command is None:
    run_kwargs = {"check": True}
else: 
    run_kwargs = {"other_option": True}
subprocess.run(command, **run_kwargs)  # raise a subprocess-run-check
```
I'm not sure if we can fix the simple case without false positive on the control flow case (?)
